ott/core/initializers.py

```diff
@@ -0,0 +1,260 @@
+    """Initialization for Sinkhorn potential/ scaling f_u."""
+
+  @abstractmethod
+  def init_dual_b(
+      self, ot_problem: linear_problems.LinearProblem, lse_mode: bool
+  ) -> jnp.ndarray:
+    """Initialization for Sinkhorn potential/ scaling g_v."""
+
+
+class DefaultInitializer(SinkhornInitializer):
+  """Default Initialization of Sinkhorn dual potentials/ primal scalings."""
+
+  def init_dual_a(
+      self, ot_problem: linear_problems.LinearProblem, lse_mode: bool
+  ) -> jnp.ndarray:
+    """Initialization for Sinkhorn potential/ scaling f_u.
+
+    Args:
+      ot_problem: OT problem between discrete distributions of size n and m.
+      lse_mode: Return potential if true, scaling if false.
+
+    Returns:
+      potential/ scaling, array of size n
+    """
+    a = ot_problem.a
+    init_dual_a = jnp.zeros_like(a) if lse_mode else jnp.ones_like(a)
+    return init_dual_a
+
+  def init_dual_b(
+      self, ot_problem: linear_problems.LinearProblem, lse_mode: bool
+  ) -> jnp.ndarray:
+    """Initialization for Sinkhorn potential/ scaling g_v.
+
+    Args:
+      ot_problem: OT problem between discrete distributions of size n and m.
+      lse_mode: Return potential if true, scaling if false.
+
+    Returns:
+      potential/ scaling, array of size m
+    """
+    b = ot_problem.b
+    init_dual_b = jnp.zeros_like(b) if lse_mode else jnp.ones_like(b)
+    return init_dual_b
+
+
+class GaussianInitializer(DefaultInitializer):
+  """GaussianInitializer.
+
+  From :cite:`thornton2022rethinking:22`.
+  Compute Gaussian approximations of each pointcloud, then compute closed from
+  Kantorovich potential betwen Gaussian approximations using Brenier's theorem
+  (adapt convex/ Brenier potential to Kantorovich). Use this Gaussian potential to
+  initialize Sinkhorn potentials/ scalings.
+
+  """
+
+  def __init__(self):
+    super().__init__()
+
+  def init_dual_a(
+      self,
+      ot_problem: linear_problems.LinearProblem,
+      lse_mode: bool,
+  ) -> jnp.ndarray:
+    """Gaussian init function.
+
+    Args:
+      ot_problem: OT problem description with geometry and weights.
+      lse_mode: Return potential if true, scaling if false.
+
+    Returns:
+      potential/ scaling f_u, array of size n.
+    """
+    # import Gaussian here due to circular imports
+    from ott.tools.gaussian_mixture import gaussian
+
+    assert isinstance(
+        ot_problem.geom, pointcloud.PointCloud
+    ), "Gaussian initializer valid only for PointCloud geom"
+
+    x, y = ot_problem.geom.x, ot_problem.geom.y
+    a, b = ot_problem.a, ot_problem.b
+
+    gaussian_a = gaussian.Gaussian.from_samples(x, weights=a)
+    gaussian_b = gaussian.Gaussian.from_samples(y, weights=b)
+    # Brenier potential for cost ||x-y||^2/2, multiply by two for ||x-y||^2
+    f_potential = 2 * gaussian_a.f_potential(dest=gaussian_b, points=x)
+    f_potential = f_potential - jnp.mean(f_potential)
+    f_u = f_potential if lse_mode else ot_problem.scaling_from_potential(
```

Comment:
`ot_problem` has no attribute `scaling_from_potential`, should be `ot_problem.geom.scaling_from_potential`.
Could you please add a test where `lse_mode=False`?

Reply:
added!